coordinate/coordinator.go

```
@@ -0,0 +1,301 @@
+		c.Registered = true
+	}
+	return err
+}
+
+// Unregister will remove the path to the given client
+func (c *Client) Unregister() error {
+	c.Close()
+	_, err := c.client.Delete(rootDir+c.flota+c.id, false)
+	return err
+}
+
+// GetConfig is to be used later for shared configuration
+func (c *Client) GetConfig() string {
+	resp, err := c.client.Get(rootDir+c.flota+config, false, false)
+
+	if resp == nil || err != nil {
+		return ""
+	}
+
+	return resp.Node.Value
+}
+
+// Watch is to allow the cluster to respond to common events from the cluster
+func (c *Client) Watch(prefix string, alert WatchFunc) {
+	watchChan := make(chan *etcd.Response)
+	stopWatch := make(chan bool)
+
+	c.watchLock.Lock()
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+
+		stopChan := make(chan bool)
+		go c.client.Watch(prefix, 0, false, watchChan, stopChan)
+		wg.Done()
+		select {
+
+		case response := <-watchChan:
+			go alert(response)
+		case stop := <-stopWatch:
+			stopChan <- stop
+			return
+		}
+	}()
+
+	wg.Wait()
+	c.watchList[prefix] = alert
+	c.stopList[prefix] = stopWatch
+	c.watchLock.Unlock()
+}
+
+// Close will send out on all the stop channels to close down any watch goroutines
+func (c *Client) Close() {
+
+	c.watchLock.Lock()
+	for _, v := range c.stopList {
+		v <- true
+	}
+	c.watchLock.Unlock()
+}
+
+// NewSimpleCoordinator returns a Client for the Client and Daemon to use to
+// register themselves with etcd. Address should the http addressable address
+// of where the etcd server is running
+//
+// Note: flota should be 'unique' as the cluster creation process will remove
+// any existing nodes and their children to ensure the test run starts with a
+// clean slate.
+func NewSimpleCoordinator(address, flota string) *Client {
+
+	client := etcd.NewClient([]string{address})
+
+	// Panic if we cannot connect to the coordination service, the daemon will
+	// not be able to participate in the test run.
+	if client == nil {
+		panic(ErrorUnableToConnect)
```

Would rather bubble up an error than panic.